src/confd/src/ietf-routing.c

```diff
@@ -0,0 +1,149 @@
+
+#include <srx/common.h>
+#include <srx/lyx.h>
+#include <srx/srx_module.h>
+#include <srx/srx_val.h>
+
+#include "core.h"
+
+#define XPATH_BASE_    "/ietf-routing:routing"
+#define XPATH_STATIC_ROUTES_    XPATH_BASE_"/control-plane-protocols/control-plane-protocol"
+#define FRR_CONF   "/etc/frr/frr.conf"
+#define FRR_CONF_NEXT FRR_CONF"+"
+#define FRR_CONF_PREV FRR_CONF"-"
+#define FRR_STATIC_CONFIG "! Generated by Infix\n\
+frr defaults traditional\n \
+hostname Router \n\
+password zebra \n \
+enable password zebra  \n \
+log syslog informational \n"
+
+static int parse_route(struct lyd_node *parent, FILE *fp) {
+	const char *outgoing_interface, *next_hop_address, *special_next_hop, *destination_prefix;
+	struct lyd_node *next_hop;
+	destination_prefix = lydx_get_cattr(parent, "destination-prefix");
+	next_hop = lydx_get_child(parent, "next-hop");
+	outgoing_interface = lydx_get_cattr(next_hop, "outgoing-interface");
+	next_hop_address = lydx_get_cattr(next_hop, "next-hop-address");
+	special_next_hop = lydx_get_cattr(next_hop, "special-next-hop");
```

Comment:
Functionally I have no real comments, but stylistically we need to talk :-D

I've just pushed PR #209 that I hope can help us avoid future style discussions.  Let's continue there, and let's involve @wkz as well, because even though I've taken the Linux kernels `.clang-format` file, I've added a few tweaks per my own preferences for discussion.